src/main/java/seedu/bigpp/command/buildercommand/BuilderListComponentCommand.java

```diff
@@ -20,27 +35,127 @@ public BuilderListComponentCommand(String arguments) {
      */
     @Override
     public String executeCommand(DataStorage dataStorage) throws BuilderMissingListException,
-            BuilderIncorrectComponentException, BuilderMissingComponentException {
+            BuilderIncorrectComponentException, BuilderMissingComponentException, BuilderInvalidNumberFlags,
+            BuilderInvalidPriceType, BuilderInvalidPriceNumber, BuilderPriceLogicException,
+            BuilderPriceMissingArguments, BuilderMissingNameException, BuilderPriceFlagException {
 
         assert UI.getUiState() == UIState.PCBUILDER : "UI state should be PCBUILDER";
 
-        String componentTypeString = getArguments();
-        if (componentTypeString.equals("")) {
+        String userInputString = getArguments();
+        userInputString = userInputString.toLowerCase();
+
+        String[] userInputStringArray = userInputString.split(" ");
+        String componentType = userInputStringArray[0];
+
+        if (userInputString.equals("")) {
             throw new BuilderMissingComponentException();
         }
-        if (!dataStorage.stringToComponentListMap.containsKey(componentTypeString)) {
+        if (!dataStorage.stringToComponentListMap.containsKey(componentType)) {
             throw new BuilderIncorrectComponentException();
         }
 
-        componentTypeString = componentTypeString.toLowerCase();
+        ComponentList<?> componentList = dataStorage.stringToComponentListMap.get(componentType);
+        ArrayList<String> flagDescriptionArray = new ArrayList<String>();
+        for (int i = 1; i < userInputStringArray.length; i++) {
+            String flagCommand = userInputStringArray[i].trim();
 
-        // throw exception if component type is not valid eg. "list jfk"
-        if (!dataStorage.stringToComponentListMap.containsKey(componentTypeString)) {
-            throw new BuilderIncorrectComponentException();
+            switch (flagCommand) {
+            case NAME_FLAG:
+                try {
+                    if (userInputStringArray.length < i + 2) {
+                        throw new BuilderMissingNameException();
+                    }
+                    String flagArgument = userInputStringArray[i + 1].trim();
+                    componentList = filterByName(componentList, flagArgument);
+                    flagDescriptionArray.add("name: " + flagArgument);
```

Comment:
If someone types `list cpu -name -price`, hence skipping the name, does "-price" get treated as the name in question?

Reply:
yes